src/convertToObject.js

```diff
@@ -3,16 +3,30 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const sources = sourceString.split(';')
+    .map(
+      string => string.trim().split(':')
+        .map(str => str.trim()),
+    );
+
+  return sources.reduce((styleObject, style) => {
+    styleObject[style[0]] = style[1];
+
+    delete styleObject[''];
```

Comment:
```suggestion
    if (style[0] && style[1]) {
      styleObject[style[0]] = style[1];
    }
```
You can add properties only if they exist instead of deleting every time